### PR TITLE
added config and logging

### PR DESCRIPTION
**Pull request by nitzantomer**

### CODE REVIEW

File: scripts/src/logging.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import * as winston from "winston";
+import {GenericTextTransportOptions} from "winston";
+import {GenericTransportOptions} from "winston";
+import {FileTransportOptions} from "winston";
+
+export interface LogTarget {
+	name: string;
+	type: "console" | "file";
+	level: "debug" | "info" | "warn" | "error";
+	timestamp?: boolean | (() => string | boolean);
```
Comment (doodyparizada) [inline scripts/src/logging.ts]: I don't understand this. is timestamp a boolean?

File: scripts/src/logging.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+export function getLogger(...targets: LogTarget[]): winston.LoggerInstance {
+	const options: winston.LoggerOptions = {};
+	options.transports = targets.map(target => createTarget(target));
+
+	return new winston.Logger(options);
+}
+
+type WinstonTransportOptions = GenericTransportOptions & GenericTextTransportOptions;
+function createTarget(target: LogTarget): winston.TransportInstance {
+	let cls: { new (options: WinstonTransportOptions): winston.TransportInstance };
```
Comment (doodyparizada) [inline scripts/src/logging.ts]: I don't know what this means

File: scripts/src/logging.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+			break;
+
+		default:
+			throw new Error("unsupported log target type: " + target.type);
+	}
+
+	return new cls(mergeOptions(defaults, options));
+}
+
+function mergeOptions(defaults: WinstonTransportOptions, options: WinstonTransportOptions): WinstonTransportOptions {
```
Comment (doodyparizada) [inline scripts/src/logging.ts]: Is there a better way?

File: scripts/src/logging.ts (hunk 1 above)
Comment (nitzantomer) [reply scripts/src/logging.ts]: `timestamp` can be either a boolean or a function which returns either a boolean or a string.

File: scripts/src/logging.ts (hunk 2 above)
Comment (nitzantomer) [reply scripts/src/logging.ts]: `cls` is a class, that is, a constructor function which expects one param (`options`) and returns a new instance of `winston.TransportInstance`.
notice that later there's:
`new cls(options)`

File: scripts/src/logging.ts (hunk 3 above)
Comment (nitzantomer) [reply scripts/src/logging.ts]: a better way?
what's wrong with this one?

File: scripts/src/logging.ts (hunk 1 above)
Comment (doodyparizada) [reply scripts/src/logging.ts]: why is it a boolean? why not string or time in seconds from epoch?

Comment (nitzantomer) [reply scripts/src/logging.ts]: it is only to set whether or not a timestamp should be included.
if so, the timestamp is added.  it's not the format.

File: scripts/src/logging.ts (hunk 3 above)
Comment (doodyparizada) [reply scripts/src/logging.ts]: a more generic, merge 2 objects

**Pull request merged**